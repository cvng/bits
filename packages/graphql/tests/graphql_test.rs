mod common;

use crate::common::execute;
use crate::common::Context;
use crate::common::TestToken;
use graphql_client::GraphQLQuery;
#[cfg(test)]
use insta::assert_json_snapshot;
use test_context::test_context;
use tokio::test;

#[derive(GraphQLQuery)]
#[graphql(
  schema_path = "../../docs/schema.gql",
  query_path = "tests/operations.graphql"
)]
pub struct BidMutation;

#[test_context(Context)]
#[test]
async fn test_bid_mutation(ctx: &mut Context) {
  let response = execute(
    ctx,
    TestToken::bidder_token(),
    BidMutation::build_query(bid_mutation::Variables {}),
  )
  .await
  .unwrap();

  assert_json_snapshot!(response, { ".data.bid.bid.id" => "[uuid]" });
}

#[derive(GraphQLQuery)]
#[graphql(
  schema_path = "../../docs/schema.gql",
  query_path = "tests/operations.graphql"
)]
pub struct StartMutation;

#[test_context(Context)]
#[test]
async fn test_start_mutation(ctx: &mut Context) {
  let response = execute(
    ctx,
    TestToken::seller_token(),
    StartMutation::build_query(start_mutation::Variables {}),
  )
  .await
  .unwrap();

  assert_json_snapshot!(response, { ".data.start.show.id" => "[uuid]" });
}
