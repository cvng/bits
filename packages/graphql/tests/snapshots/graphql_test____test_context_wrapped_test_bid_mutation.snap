---
source: packages/graphql/tests/graphql_test.rs
expression: response
---
{
  "data": {
    "bid": {
      "bid": {
        "id": "[uuid]",
        "auctionId": "00000000-0000-0000-0000-000000000000",
        "bidderId": "00000000-2000-0000-0000-000000000000",
        "amount": "500"
      }
    }
  }
}
